mathesar/reflection.py

```diff
@@ -42,53 +44,101 @@ def reflect_databases():
 
 
 # TODO creating a one-off engine is expensive
-def reflect_schemas_from_database(database_name):
-    engine = create_mathesar_engine(database_name)
+def reflect_schemas_from_database(database):
+    engine = create_mathesar_engine(database.name)
     db_schema_oids = {
         schema['oid'] for schema in get_mathesar_schemas_with_oids(engine)
     }
 
-    database = models.Database.current_objects.get(name=database_name)
+    schemas = []
     for oid in db_schema_oids:
-        schema, boolean = models.Schema.current_objects.get_or_create(oid=oid, database=database)
-    for schema in models.Schema.current_objects.all():
+        schema = models.Schema(oid=oid, database=database)
+        schemas.append(schema)
+    models.Schema.current_objects.bulk_create(schemas, ignore_conflicts=True)
+    deleted_schemas = []
+    for schema in models.Schema.current_objects.all().select_related('database'):
         if schema.database.name == database and schema.oid not in db_schema_oids:
```

Comment:
Sort of unrelated, but notice that we're checking for equality between a database's name and a database model instance here. Fixed this in my performance pr, but might want to change it here as well, in case that fix gets lost in merge conflicts.

Reply:
Thanks for the catch!